File: REST_DotNET_Coffee_Android/DTO/OrderDTO/OrderResponseDTO.cs
#nullable disable

public class OrderResponseDTO
{
    public String PaymentMethod {  get; set; }

    public string UserName { get; set; }

    public List<OrderItemResponseDTO> ItemOrderList { get; set; }

    public double TotalPrice {
        get
        {
            return ItemOrderList?.Sum(item => item.TotalPrice) ?? 0;
        }
    }
}

File: Mobile_Coffee/Project-Mobile__Coffe-Store/app/src/main/java/com/nlu/packages/dotnet_callapi/responsedto/OrderResponseDTO.java
package com.nlu.packages.dotnet_callapi.responsedto;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class OrderResponseDTO {
    private String paymenMethod;
    private String userName;
    private List<OrderItemResponseDTO> itemOrderList;
    public double getTotalPrice() {
        double result =0;
        for (OrderItemResponseDTO oir : itemOrderList) {
            result += oir.getTotalPrice();
        }
        return result;
    }
}
